src/main/java/mate/academy/security/jwt/JwtTokenProvider.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.security.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import jakarta.servlet.http.HttpServletRequest;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+import mate.academy.exception.InvalidJwtAuthenticationException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+@Component
+@PropertySource("classpath:application.properties")
+public class JwtTokenProvider {
+    @Value("${security.jwt.token.secret-key:secret}")
+    private String secretKey;
```

Comment:
should we encode this key in `@PostConstruct`?

Reply:
as i see all don't realize at start need @PostConstruct method for  encode in base64 of secret key 
thus this must be not optional task in learning process in future imho
refactored